packages/loopring_v3/contracts/impl/libexchange/ExchangeBlocks.sol

```diff
@@ -59,227 +60,115 @@ library ExchangeBlocks
         uint8 previousMakerFeeBips
     );
 
-    function commitBlock(
-        ExchangeData.State storage S,
-        uint8  blockType,
-        uint16 blockSize,
-        uint8  blockVersion,
-        bytes  calldata data,
-        bytes  calldata /*offchainData*/
-        )
-        external
-    {
-        commitBlockInternal(
-            S,
-            ExchangeData.BlockType(blockType),
-            blockSize,
-            blockVersion,
-            data
-        );
-    }
+    event BlockFeeWithdrawn(
+        uint    indexed blockIdx,
+        uint            amountRewarded,
+        uint            amountFined
+    );
 
-    function verifyBlocks(
+    function submitBlocks(
         ExchangeData.State storage S,
-        uint[] calldata blockIndices,
-        uint[] calldata proofs
+        ExchangeData.Block[] memory blocks,
+        address payable feeRecipient
         )
-        external
+        public
     {
```

Comment:
It will depend on how expensive the transfers are. If the withdrawals are in ETH than that will be cheaper and more withdrawals in a block will be possible.

In V3.0 we would have to pay ~4,500 gas/withdrawal just to store the data so it was available after the block is finalized. And then we'd still have to do the transfer  (+ some gas refunds for reverting the storage we used to store the withdrawal data back to 0 after doing the transfer).

256 withdrawals/block are easily possible. Even 512 withdrawals/block will likely be possible because they won't all consume 20,000 gas/withdrawal.